From c2e5f121e180d7eb66a8dafb10a0d6f50f301a4f Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:59 -0400
Subject: [misc] mei: drop redundant list_del_init

Message-id: <1399041470-21330-27-git-send-email-prarit@redhat.com>
Patchwork-id: 79313
O-Subject: [RHEL7.1 PATCH BZ 1088953 26/77] mei: drop redundant list_del_init
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 0f1d4ce5d0d09de5db92df7f51cdf7163bbb61f9
Author: Alexander Usyskin <alexander.usyskin@intel.com>
Date:   Mon Nov 11 13:26:07 2013 +0200

    mei: drop redundant list_del_init

    list_del_init appears twice in row in mei_cl_unlink, drop one.

    Signed-off-by: Alexander Usyskin <alexander.usyskin@intel.com>
    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Cc: Dean Nelson <dnelson@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/client.c b/drivers/misc/mei/client.c
index 87c96e4..6184e85 100644
--- a/drivers/misc/mei/client.c
+++ b/drivers/misc/mei/client.c
@@ -344,8 +344,6 @@ int mei_cl_unlink(struct mei_cl *cl)
 
  cl->state = MEI_FILE_INITIALIZING;
 
- list_del_init(&cl->link);
-
  return 0;
 }
 
-- 
1.7.1